4-sinatra/app.rb

```diff
@@ -0,0 +1,41 @@
+require 'bundler'
+
+Bundler.require(:default)
+
+require "sinatra"
+require 'sinatra/reloader'
+require 'fast_secure_compare/fast_secure_compare'
+
+enable :sessions
+
+get '/' do
+  if session[:logged]
+    redirect to '/login'
+  else
+    erb :form
+  end
+end
+
+post '/login' do
+  if session[:logged]
+    erb :logged
+  else
+    pass_array = File.open("password.txt").read.split(":")
```

Comment:
Gdyby aplikacja nie dawała opcji zmiany hasła to można by ten plik czytać tylko raz i przechowywać wartość hasła w stałej. W tym przypadku jednak jeśli chcemy wspierać zmianę hasła w locie przez skrypt w tle działające aplikacji faktycznie trzeba do pliku za każdym razem.